src/import/import-compress.c

```diff
@@ -231,6 +290,55 @@ int import_compress_init(ImportCompress *c, ImportCompressType t) {
                 break;
 #endif
 
+#if HAVE_ZSTD
+        case IMPORT_COMPRESS_ZSTD: {
+                size_t r2;
+                int ncpus;
+
+                assert(c->zstd_c == NULL);
+                c->zstd_c = ZSTD_createCStream();
+                if (c->zstd_c == NULL)
+                        return -ENOMEM;
+
+                /* TODO: better default? zstd -3 is really weak */
+                r2 = ZSTD_CCtx_setParameter(c->zstd_c, ZSTD_c_compressionLevel, ZSTD_CLEVEL_DEFAULT);
+                if (ZSTD_isError(r2))
+                        return log_error_errno(
+                                SYNTHETIC_ERRNO(EIO),
+                                "Failed to set zstd compression level: %s",
+                                ZSTD_getErrorName(r2)
+                        );
+
+                r2 = ZSTD_CCtx_setParameter(c->zstd_c, ZSTD_c_checksumFlag, 1);
+                if (ZSTD_isError(r2))
+                        return log_error_errno(
+                                SYNTHETIC_ERRNO(EIO),
+                                "Failed to enable zstd output checksumming: %s",
+                                ZSTD_getErrorName(r2)
+                        );
+
+                ncpus = cpus_in_affinity_mask();
+                if (ncpus > 0) {
+                        r2 = ZSTD_CCtx_setParameter(c->zstd_c, ZSTD_c_nbWorkers, ncpus);
```

Comment:
if we enable this, will we lose reproducibility?

Reply:
I don't think we ever had it. E. g. gzip-compressed outputs are not reproducible here.